### PR TITLE
Update to store objectives and constraints as optimizationProblem properties

### PR DESCRIPTION
**Pull request by tobiasbecher**

### CODE REVIEW

**Review by tobiasbecher — COMMENTED** [review]
There are some changes related to normalization of objectives, that are not relevant for this part, but for the sandwich algorithms. Let me know, if I should move that to a seperate pull request

File: matRad_fluenceOptimization.m [hunk 1]
```diff
@@ -249,7 +249,7 @@
 backProjection.scenarios    = ixForOpt;
 backProjection.scenarioProb = pln.multScen.scenProb;
 
-optiProb = matRad_OptimizationProblem(backProjection);
+optiProb = matRad_OptimizationProblem(backProjection,cst);
```
Comment (wahln) [inline matRad_fluenceOptimization.m:252]: See comment in the OptimizationProblem file!

File: optimization/@matRad_OptimizationProblem/matRad_OptimizationProblem.m [hunk 2]
```diff
@@ -24,17 +24,27 @@
     
     properties
         BP
+        normalizationScheme = struct('type','none');
```
Comment (wahln) [inline optimization/@matRad_OptimizationProblem/matRad_OptimizationProblem.m:27]: What does "type" stand for? Not very instructive

File: optimization/@matRad_OptimizationProblem/matRad_OptimizationProblem.m [hunk 3]
```diff
@@ -24,17 +24,27 @@
...
+        objectives = {}; %cell array storing all objectives, has to be initialized at the start
+        constraints = {}; %
+        objidx;
```
Comment (wahln) [inline optimization/@matRad_OptimizationProblem/matRad_OptimizationProblem.m:30]: would adhere to coding style by using capital I: objIdx, constrIdx

File: optimization/@matRad_OptimizationProblem/matRad_OptimizationProblem.m [hunk 4]
```diff
@@ -24,17 +24,27 @@
...
 
         minimumW = NaN;
         maximumW = NaN;
     end
+
     
     methods
-        function obj = matRad_OptimizationProblem(backProjection)
-            obj.BP = backProjection;
+        function obj = matRad_OptimizationProblem(backProjection,cst)
```
Comment (wahln) [inline optimization/@matRad_OptimizationProblem/matRad_OptimizationProblem.m:42]: I am not sure if we should add the cst as a parameter (compatibility to other stuff).
We could also generate this cache the first time we need it instead?

Comment (wahln) [reply optimization/@matRad_OptimizationProblem/matRad_OptimizationProblem.m:42]: But actually I am not sure what's the best strategy here.... maybe leave it like this.

File: optimization/@matRad_OptimizationProblem/matRad_constraintJacobian.m [hunk 5]
```diff
@@ -59,154 +59,138 @@
...
+            jacobSub = constraint.computeDoseConstraintJacobian(d_i);
             
-        
+         otherwise
+            matRad_cfg.dispError('Robustness setting %s not yet supported!',constraint.robustness);
+      end
+      
+      nConst = size(jacobSub,2);
+      
+      %Iterate through columns of the sub-jacobian
```
Comment (wahln) [inline optimization/@matRad_OptimizationProblem/matRad_constraintJacobian.m:134]: Didn't you want to move this to the projections or was this other work?

File: optimization/@matRad_OptimizationProblem/matRad_objectiveFunction.m [hunk 6]
```diff
@@ -1,221 +1,95 @@
...
+    %
+    % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+
+    fIndv = matRad_objectiveFunctions(optiProb,w,dij,cst);
+    %fIndv should be an mxn matrix where m is the number of objectives and n should be the number of scenarios used
+
+    switch optiProb.normalizationScheme.type
+        case 'UL' %does not make sense together with robustness or scenarios? -> Add check?
+            fIndv = optiProb.normalizeObjectives(fIndv')';%ISSUES IF COWC is used -
```
Comment (wahln) [inline optimization/@matRad_OptimizationProblem/matRad_objectiveFunction.m:39]: Are these issues solved?

Comment (wahln) [reply optimization/@matRad_OptimizationProblem/matRad_objectiveFunction.m:39]: And what issues?

File: optimization/@matRad_OptimizationProblem/matRad_objectiveFunction.m [hunk 7]
```diff
@@ -1,221 +1,95 @@
...
+    end
 
-% compute objective function for every VOI.
-for  i = 1:size(cst,1)
+    %TODO: sum up over scenarios
```
Comment (wahln) [inline optimization/@matRad_OptimizationProblem/matRad_objectiveFunction.m:42]: What's with this TODO?

File: optimization/@matRad_OptimizationProblem/matRad_objectiveFunction.m
Comment (wahln) [inline optimization/@matRad_OptimizationProblem/matRad_objectiveFunction.m:1]: This file seems to be not very finalized, maybe clarify the comments and if something is still debatable.

**Review by wahln — CHANGES_REQUESTED** [review]
Looks good, I am still not convinced that everything is done given the state of a few files.
Another comment: if we have an objectiveFunctions file that stores objective values as array, why do we not have a objectiveJacobian?

**Comment by tobiasbecher** [comment]
Closed due to changes being added in other pull request

**Pull request closed without merging**